src/handlers/int-test-support/helpers/commonHelpers.ts

```diff
@@ -54,6 +54,54 @@ export const waitForTrue = async (
   });
 };
 
+export const poll = async <Resolution>(
+  promise: () => Promise<Resolution>,
+  completionCondition: (resolution: Resolution) => boolean,
```

Comment:
Note the generic `<Resolution>` gives you type safety when writing your `completionCondition` function